clang/lib/Frontend/ASTUnit.cpp

```diff
@@ -1752,6 +1752,12 @@ ASTUnit *ASTUnit::LoadFromCommandLine(
     IntrusiveRefCntPtr<llvm::vfs::FileSystem> VFS) {
   assert(Diags.get() && "no DiagnosticsEngine was provided");
 
+  // If no VFS was provided, create one that tracks the physical file system.
+  // If '-working-directory' was passed as an argument, 'createInvocation' will
+  // set this as the current working directory of the VFS.
+  if (!VFS)
```

Comment:
Why is this done here instead of above the first use of VFS to create a FileManager above?

Reply:
Hmm, could you elaborate? This is the first use of `VFS` in the function. Or do you mean why are we doing it here instead of where we were previously doing it (immediately before creating the FileManager)? We're doing it here so that it can be used by `createInvocation`, such that `-working-directory` will set the working directory of the VFS.

Reply:
Sorry, I just confused myself when looking at the rest of the file and lost track of where exactly this change lined up.